examples/hal/quad/main.rs

```diff
@@ -3,614 +3,1380 @@
     allow(dead_code, unused_extern_crates, unused_imports)
 )]
 
-extern crate env_logger;
-extern crate gfx_hal as hal;
 #[cfg(feature = "dx12")]
 extern crate gfx_backend_dx12 as back;
-#[cfg(feature = "vulkan")]
-extern crate gfx_backend_vulkan as back;
-#[cfg(feature = "metal")]
-extern crate gfx_backend_metal as back;
 #[cfg(feature = "gl")]
 extern crate gfx_backend_gl as back;
+#[cfg(feature = "metal")]
+extern crate gfx_backend_metal as back;
+#[cfg(feature = "vulkan")]
+extern crate gfx_backend_vulkan as back;
+#[cfg(not(any(feature = "vulkan", feature = "dx12", feature = "metal", feature = "gl")))]
+extern crate gfx_backend_empty as back;
 
-extern crate winit;
+extern crate env_logger;
+extern crate gfx_hal as hal;
+extern crate glsl_to_spirv;
 extern crate image;
+extern crate takeable_option;
```

Comment:
do we still need that? I thought we had a consensus that laying down everything into a function body is fine for the example